### DISCUSSION TITLE
Time filters not being translated, even with appropriate .json, .po & .mo files.

### DISCUSSION THREAD

**Discussion by Ralkion** [apache/superset#25994] [Q&A / Help] [answered]
Hello all,

I've been struggling with the translations for a couple of days now and I think I've resolved all my issues except these two. The value displayed in the Time Range filter after selecting a translated option is displayed in English.
![image](https://github.com/apache/superset/assets/90489859/d78be8f7-d695-4efc-aa40-f4678a62160a)

The second issue I've run into is that the Time Grain options are all displayed in English.
![image](https://github.com/apache/superset/assets/90489859/cde0c38a-5961-4d89-aca8-fe685f0bab53)

I've read lots of discussions & MRs around translations - particularly those done by AndVK and hbruch. The most relevant is where hbruch says "No filter for many charts still is not localized." in this issue: https://github.com/apache/superset/issues/17779.

I also found this issue, https://github.com/apache/superset/issues/25728 complaining about the Time Range, and a comment saying that it's fixed in 3.0.0. However when I tried out 3.0.0 I found no changes to the problem outlined in that issue.

Is there something that I've missed in my configuration or are these two not actually translated in the UI?

**Comment by aehanno** [comment, marked as answer]
In the https://github.com/apache/superset/issues/25728, that just correct the translation inside the time range filter when you choose the time range.
I try to find a solution on the display inside the filter tab but I didn't figure out a solution

**Reply by Ralkion** [reply]
Thanks! I'm glad I'm not the only one having this issue. I've been struggling with it for a week convinced I must be doing something wrong.

I've got a PR coming soon that will address both of these issues. I'm just trying to get through some of the unit test failures now.

**Reply by aehanno** [reply]
Oh great, do you have the link of this PR, I can try to check if I can do something on the test

**Comment by Stefan781** [comment]
We ran into the missing translation for the time grain options. Is there any progress?
Edit: There is a [PR](https://github.com/apache/superset/pull/26074). Thanks for your work!
